### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by vboychyk**

### CODE REVIEW

**Comment by shablenko-sednor** [comment]
[XS - Sample full-stack app](https://trello.com/c/buIZPahz/3851-xs-sample-full-stack-app)

File: routes/index.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+"use strict";
+
+const os = require("os");
+const fs = require("fs");
+const path = require("path");
+const express = require("express");
+const stream = require("stream");
+const Handlebars = require("handlebars");
+const wkhtmltopdf = require("wkhtmltopdf");
```
Comment (dhurlburtusa) [inline routes/index.js:9]: I was reading about the `wkhtmltopdf` npm package. Sounds like you still need to have the command line tool installed and set on the PATH. If this is true to get this project to work, then let's document it with the other installation/setup documentation.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "example-full-stack",
+  "version": "1.0.0",
+  "private": false,
+  "description": "XcooBee SDK and Payment SDK usage example",
+  "main": "server.js",
+  "scripts": {
+    "start": "nodemon server.js",
+    "dev": "node server.js"
```
Comment (dhurlburtusa) [inline package.json:9]: This is the first project I've seen that has both a `start` and a `dev` script. In the projects I've looked at, one or the other is used and it is used for development purposes. That is, it is monitoring changes and refreshing the app automatically.

So, I am thinking is might be better to rename `dev` to something like `serve`. Thoughts?

File: public/javascripts/index.js [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+                    return;
+                }
+
+                const properties = {};
+                const formData = $("#invoice_form").serializeArray();
+                formData.forEach((item) => {
+                    properties[item.name] = item.value;
+                });
+
+                window.open(`/generate-invoice?ref=${properties.payment_reference}&amount=${properties.payment_amount}&campaignId=${settings.campaign_id}`, "_blank");
```
Comment (dhurlburtusa) [inline public/javascripts/index.js:266]: To be on the safe side, I would use `encodeUriComponet` for all these parameters even if we know they are safe ahead of time.

File: views/layout.hbs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  <title>{{title}}</title>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel='icon' href='/images/favicon.png' class='js-favicon'>
+  <link rel="stylesheet"
+        href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/css/bootstrap.min.css"
+        integrity="sha384-9aIt2nRpC12Uk9gS9baDl411NQApFmC26EwAOH8WgZl5MYYxFfc+NcPb1dKGj7Sk"
+        crossorigin="anonymous">
+  <link rel='stylesheet' href='/stylesheets/style.css' />
+  <script src='https://app.xcoobee.net/scripts/sdk/openpgp.min.js' />
```
Comment (dhurlburtusa) [inline views/layout.hbs:13]:
```suggestion
  <script src='https://app.xcoobee.net/scripts/sdk/openpgp.min.js'></script>
```

File: views/settings.hbs [hunk 5]
```diff
@@ -0,0 +1,71 @@
+<div class="alert alert-secondary mb-3" role="alert">
+  Please copy the values from your XcooBee account or from your downloaded key files.
+</div>
+<form action="#" class="mt-3 settings-form">
+  <div class="row m-0">
+    <div class="col-sm-4 p-2">
+      <h6>Needed for Polling</h6>
+      <a href="https://testapp.xcoobee.net/user/settings/security">Open XcooBee API Settings</a>
```
Comment (dhurlburtusa) [inline views/settings.hbs:8]: I am not sure what the correct answer is but do we want to use the test system or the real production system?

Note: If we change to use the real prod system, there are 2 other references below.

In the README.md file there is a Prerequisites section that states you will need a XcooBee account. If it should be an account on the test system, then let's update the documentation to clarify that.

File: public/javascripts/index.js [hunk 6]
```diff
@@ -0,0 +1,269 @@
+/* eslint-disable */
+
+const COOKIE_NAME = "settings";
```
[the remark this answers is not in the document]
Comment (shablenko-sednor) [reply public/javascripts/index.js:3]: this project will be used by other developers, we don`t need to care about browser support here

[the remark this answers is not in the document]
Comment (dhurlburtusa) [reply public/javascripts/index.js:3]: Sounds good.

**Pull request merged**